watcher.py

```diff
@@ -27,13 +30,21 @@ def __init__(self, upstream, user, token, db_path, build, log_path='log.json', i
         self.test = 'wpt'
         self.log_path = log_path
         self.is_dummy = is_dummy
+        self.remote = remote
+        self.branch = branch
+        self.subdir = subdir
+        self.suite = suite
+        self.no_update = no_update
+        self.no_execute = no_execute
         if is_dummy:
             print '\033[1m\033[93mRunning in dummy mode: API will not be used!\033[0m'
         if os.path.exists(log_path):
             with open(log_path, 'r') as fd:
                 self.results = json.load(fd)
+        else:
+            self.results = {}
```

Comment:
Nit: This should be moved to the top along with the other declarations. Let's override if the log exists already.